container.te

```diff
@@ -1033,3 +1033,33 @@ dontaudit container_domain device_node:chr_file setattr;
 dontaudit container_domain sysctl_type:file write;
 
 allow container_t proc_t:filesystem remount;
+
+
+########################################
+#
+# virt_launcher_t local policy
+#
+# SELinux security policy for kubevirt project
+# Temporary solution for rhbz#1795975
+
+type virt_launcher_t, container_domain;
+domain_type(virt_launcher_t)
+
+allow virt_launcher_t self:tun_socket { attach_queue relabelfrom relabelto create_socket_perms };
+
+manage_dirs_pattern(virt_launcher_t, container_var_run_t, container_var_run_t)
+manage_files_pattern(virt_launcher_t, container_var_run_t, container_var_run_t)
+
+manage_dirs_pattern(virt_launcher_t, container_share_t, container_share_t)
+manage_files_pattern(virt_launcher_t, container_share_t, container_share_t)
+
+manage_dirs_pattern(virt_launcher_t, container_runtime_tmp_t, container_runtime_tmp_t)
+manage_files_pattern(virt_launcher_t, container_runtime_tmp_t, container_runtime_tmp_t)
```

Comment:
Are these runtime files being added to the container?
Do we know what command they are executing to launch the containers?

Reply:
Some time ago we used to use /tmp/healthy for file-based health monitoring--we don't anymore. I'm wondering if this is cruft left over from that.

I'm currently verifying if there's any other consequences to removing this.

Reply:
Okay, this probably could be removed also.

Reply:
I tested removing this line and lots of tests failed. Oddly they failed on access to files such as this:

`/var/run/kubevirt-private/vmi-disks/disk0/disk.img`

Which is probably a secondary effect as that is a bindmounted elsewhere. Would it be ok to leave this rule in place?

Reply:
Follow-up: the above is a side-effect. CDI's upload server still uses /tmp/healthy to report status. Either the startup script is crashing because it can't write there, or k8s is crashlooping the pod because it never reports as healthy.

One KubeVirt component still does require read/write access to /tmp.

Reply:
Can we just label the content in /var/run/kubevert-private to container_file_t, or volume mount it into the container engine with the :Z or :z options?

Reply:
The issue with /var/run/kubevirt-private was a red herring. the real issue was that the CDI upload server failed to start when I removed permissions to write to /tmp.